File: backend/modules/sabor-familia/src/main/java/br/com/puc/saborfamilia/service/mensagem/dto/response/EventoMensagemWs.java
package br.com.puc.saborfamilia.service.mensagem.dto.response;

import br.com.puc.saborfamilia.enums.TipoEventoMensagem;
import java.time.LocalDateTime;

public record EventoMensagemWs(
  TipoEventoMensagem tipo,
  Long conversaId,
  MensagemResponse mensagem,
  String ultimaMensagem,
  LocalDateTime dataUltimaMensagem,
  Long naoLidas
) {

}

File: frontend/src/features/sabor_familia/dto/menssagem/response/EventoMensagemWs.ts
import type { MensagemResponse } from "./MensagemResponse";

export type TipoEventoMensagemWs = "NOVA" | "APAGADA";

export interface EventoMensagemWs {
  tipo: TipoEventoMensagemWs;
  conversaId: number;
  mensagem: MensagemResponse;
  ultimaMensagem: string | null;
  dataUltimaMensagem: string | null;
  naoLidas?: number | null;
}
